pkg/dockerregistry/server/pullthroughblobstore.go

```diff
@@ -179,8 +176,40 @@ func (pbs *pullthroughBlobStore) copyContent(store BlobGetterService, ctx contex
 	return desc, nil
 }
 
+// storeLocalInBackground spawns a separate thread to copy the remote blob from the remote registry to the
+// local blob store.
+// The function assumes that localBlobStore is thread-safe.
+func storeLocalInBackground(ctx context.Context, repo *repository, localBlobStore distribution.BlobStore, dgst digest.Digest) {
+	// leave only the essential entries in the context
+	newCtx := context.WithValue(context.Background(), "instance.id", ctx.Value("instance.id"))
```

Comment:
instance.id is an id of the instance, it is the same across all requests. It is changed only when the application is restarted.